shiny/express/_run.py

```diff
@@ -0,0 +1,243 @@
+from __future__ import annotations
+
+import ast
+import os
+import sys
+from pathlib import Path
+from typing import cast
+
+from htmltools import Tag, TagList
+
+from .. import render, ui
+from .._app import App
+from ..session import Inputs, Outputs, Session
+from ._recall_context import RecallContextManager
+from .display_decorator._func_displayhook import _display_decorator_function_def
+from .display_decorator._node_transformers import (
+    DisplayFuncsTransformer,
+    display_decorator_func_name,
+)
+
+__all__ = (
+    "wrap_express_app",
+    "is_express_app",
+)
+
+_DEFAULT_PAGE_FUNCTION = ui.page_fluid
+
+
+def wrap_express_app(file: Path | None = None) -> App:
+    """Wrap a Shiny express-mode app into a Shiny `App` object.
+
+    Parameters
+    ----------
+    file
+        The path to the file containing the Shiny express application. If `None`, the
+        `SHINY_EXPRESS_APP_FILE` environment variable is used.
+
+    Returns
+    -------
+    :
+        A `shiny.App` object.
+    """
+    if file is None:
+        app_file = os.getenv("SHINY_EXPRESS_APP_FILE")
+        if app_file is None:
+            raise ValueError(
+                "No app file was specified and the SHINY_EXPRESS_APP_FILE environment variable "
+                "is not set."
+            )
+        file = Path(os.getcwd()) / app_file
+
+    # TODO: title and lang
+    app_ui = ui.page_output("__page__")
+
+    def express_server(input: Inputs, output: Outputs, session: Session):
+        try:
+            dyn_ui = run_express(file)
```

Comment:
```suggestion
    app_ui = run_express(file)

    def express_server(input: Inputs, output: Outputs, session: Session):
        try:
            dyn_ui = run_express(file)
```

Reply:
The reasons for suggesting this (during live discussion):

1. HTML dependencies load with the page load which leads to smoother rendering (less fouc)

Downside:

1. Process startup is slightly slower

Reply:
Oops, also would need to get rid of render.ui below